extension/oauth2clientauthextension/extension.go

```diff
@@ -38,6 +39,14 @@ type ClientCredentialsAuthenticator struct {
 // ClientCredentialsAuthenticator implements ClientAuthenticator
 var _ configauth.ClientAuthenticator = (*ClientCredentialsAuthenticator)(nil)
 
+type errorWrappingTokenSource struct {
+	ts      oauth2.TokenSource
+	wrapper string
```

Comment:
Can we move this to module level error string template. I don't think there is a need for this here.